apps/resources/views.py

```diff
@@ -29,11 +31,33 @@ def get_form_kwargs(self):
         return kwargs
 
 
-class ResourceDetailView(DetailView):
+class ResourceDetailView(DetailView, CreateView):
     model = Resource
+    form_class = CommentForm
+    template_name = 'resources/resource_detail.html'
+
+    def get_success_url(self):
+        return self.get_object().get_absolute_url()
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         self.object.hits = F('hits') + 1
         self.object.save()
         return context
+
+    def form_invalid(self, form):
+        # To prevent from crashing form_invalid should pass resource object instead of comment
+        # empty one.
+        self.object = self.get_object()
+        return super().form_invalid(form)
+
+    def get_form_kwargs(self):
```

Comment:
Maybe change these to `get_initial` instead? It's a method used for the initial form kwarg, so is probably more appropriate!